uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/FederatedLoginIT.java

```diff
@@ -0,0 +1,347 @@
+        String spZoneUrl = baseUrl.replace("localhost", spZoneId + ".localhost");
+        RestTemplate adminClient = getAdminClient();
+        RestTemplate identityClient = getIdentityClient();
+        //Creating Orch IDP Zone
+        IdentityZone idpZone = IntegrationUtilsOrchestrator.createOrchZone(identityClient, baseUrl, idpZoneId, idpZoneId, null);
+        idpZone.getId();
+        String idpZoneUserEmail = new RandomValueStringGenerator().generate() + "@samltesting.org";
+        //Create user for IDP Admin
+        createZoneUser(idpZoneId, idpZoneUserEmail, idpZoneUrl);
+        SamlConfig samlConfig = new SamlConfig();
+        samlConfig.setWantAssertionSigned(true);
+        samlConfig.addAndActivateKey("key-1", new SamlKey(key1, passphrase1, certificate1));
+        samlConfig.addKey("key-2", new SamlKey(key2, passphrase2, certificate2));
+        IdentityZoneConfiguration config = new IdentityZoneConfiguration();
+        config.setSamlConfig(samlConfig);
+        //Creating Orch SP Zone
+        IdentityZone spZone = IntegrationUtilsOrchestrator.createOrchZone(identityClient, baseUrl, spZoneId, spZoneId, config);
+        //Get Client credentials to for SP Admin token
+        String spZoneAdminToken = IntegrationTestUtils.getClientCredentialsToken(serverRunning, "admin", "adminsecret");
+        //Get IDP meta Data
+        SamlIdentityProviderDefinition samlIdentityProviderDefinition = createZone1IdpDefinition(IDP_ENTITY_ID);
+        //configure IDP with Metadata
+        IdentityProvider<SamlIdentityProviderDefinition> idp = getSamlIdentityProvider(spZone.getId(), spZoneAdminToken, samlIdentityProviderDefinition);
+        //Get Sp meta data
+        SamlServiceProviderDefinition samlServiceProviderDefinition = createZone2SamlSpDefinition("cloudfoundry-saml-login");
+        //Configure SP with Metadata
+        SamlServiceProvider service = getSamlServiceProvider(idpZone.getId(), spZoneAdminToken, samlServiceProviderDefinition, "testzone2.cloudfoundry-saml-login", "Local SAML SP for testzone2", baseUrl);
+        //Login into SP with IDP credentials
+        performLogin(idpZone.getId(), idpZoneUserEmail, idpZoneUrl, spZone, spZoneUrl, samlIdentityProviderDefinition);
+        webDriver.get(baseUrl + "/logout.do");
+        webDriver.get(spZoneUrl + "/logout.do");
+    }
+
+    private RestTemplate getAdminClient() {
+        return IntegrationUtilsOrchestrator.getClientCredentialsTemplate(
+                IntegrationUtilsOrchestrator.getClientCredentialsResource(
+                        baseUrl, new String[0], "admin", "adminsecret")
+        );
+    }
+
+    private RestTemplate getIdentityClient() {
+        return IntegrationUtilsOrchestrator.getClientCredentialsTemplate(
+                IntegrationUtilsOrchestrator.getClientCredentialsResource(
+                        baseUrl, new String[]{"zones.write", "zones.read", "scim.zones"}, "identity", "identitysecret")
+        );
+    }
+
+
+
+
+
```

Comment:
Below methods are duplicates of methods from SamlLoginWithLocalIdpIT. Can you refactor common methods into a common class, for example, BaseSamlTest and make both SamlLoginWithLocalIdpIT and FederateLoginIT derive from BaseSamlTest?